### PR TITLE
Converting package to use Roxygen

### PR DESCRIPTION
**Pull request by JackStat**
I about went crazy on this one until I found R2roxygen. Hopefully this is useful

### CODE REVIEW

File: pkg/caret/R/caret-package.R [hunk 1]
```diff
@@ -0,0 +1,1758 @@
...
+#' for Compound Classification and QSAR Modeling, \emph{Journal of Chemical
+#' Information and Computer Sciences}, Vol. 43, pg. 1947-1958.
+#' @keywords datasets
+NULL
+
+
+
+
+
+#' A List of Available Models in train
```
Comment (topepo) [inline pkg/caret/R/caret-package.R]: The list of available models is automatically generated by a file in the `release_process` directory . Can you come up with a way to use that code to  work with Roxygen?

**Comment by topepo** [comment]
Can you modify [this file](https://github.com/topepo/caret/blob/master/release_process/make_model_Rd.R) that is used to generate the current model list? Normally it writes [an additional Rd](https://github.com/topepo/caret/blob/master/pkg/caret/man/models.Rd) file with the information.

**Comment by JackStat** [comment]
I wonder if we can use it and roxygen.  I will test it out today

**Comment by JackStat** [comment]
Rebased so it would merge easier.

**Comment by JackStat** [comment]
Do you want me to rebase this again? It is kind of a beast and conflicts with changes to any part of the code

**Comment by topepo** [comment]
Yes, it its not too much of a pain. The changes that I just checked in shouldn't have any conflicts; I just updated some model code.

I'll probably merge this in Thursday or Friday.

**Comment by topepo** [comment]
Can you sync and resubmit the PR?

**Comment by JackStat** [comment]
Is that what you were looking for?  Should merge with no conflicts now

**Comment by topepo** [comment]
Yep. That did the trick. Thanks

**Comment by topepo** [comment]
I just built the package and had these messages doing the checking 
```
* checking for missing documentation entries ... WARNING
Undocumented code objects:
  ‘plot.calibration’ ‘plot.lift’ ‘summary.bag’ ‘summary.train’
Undocumented data sets:
  ‘bbbDescr’ ‘logBBB’ ‘cars’
All user-level objects in a package should have documentation entries.
See chapter ‘Writing R documentation files’ in the ‘Writing R
Extensions’ manual.
* checking for code/documentation mismatches ... OK
* checking Rd \usage sections ... WARNING
Undocumented arguments in documentation object 'BoxCoxTrans.default'
  ‘lim’ ‘method’

Documented arguments not in \usage in documentation object 'as.table.confusionMatrix':
  ‘what’

Undocumented arguments in documentation object 'calibration'
  ‘dwidth’

Undocumented arguments in documentation object 'caret-internal'
  ‘x’ ‘y’ ‘wts’ ‘method’ ‘tuneValue’ ‘obsLevels’ ‘pp’ ‘last’ ‘sampling’
  ‘classProbs’ ‘...’ ‘modelFit’ ‘newdata’ ‘preProc’ ‘param’ ‘pred’
  ‘obs’ ‘fixed’ ‘seq’ ‘exclude’ ‘object’ ‘top’ ‘ind’ ‘names’ ‘iter’
  ‘start’ ‘testX’ ‘testY’ ‘sbfControl’ ‘data’ ‘smoother’ ‘cut’ ‘prefix’
  ‘items’

Documented arguments not in \usage in documentation object 'caretFuncs':
  ‘x’ ‘metric’ ‘maximize’ ‘tol’ ‘y’ ‘size’

Documented arguments not in \usage in documentation object 'caretSBF':
  ‘x’ ‘y’

Documented arguments not in \usage in documentation object 'confusionMatrix':
  ‘reference’ ‘positive’ ‘dnn’ ‘prevalence’ ‘mode’

Documented arguments not in \usage in documentation object 'createDataPartition':
  ‘k’ ‘returnTrain’ ‘initialWindow’ ‘horizon’ ‘fixedWindow’ ‘skip’

Documented arguments not in \usage in documentation object 'diff.resamples':
  ‘object’ ‘digits’ ‘a’ ‘b’

Documented arguments not in \usage in documentation object 'dotplot.diff.resamples':
  ‘what’

Documented arguments not in \usage in documentation object 'dummyVars':
  ‘data’ ‘sep’ ‘levelsOnly’ ‘fullRank’ ‘object’ ‘newdata’ ‘na.action’
  ‘n’ ‘contrasts’ ‘sparse’ ‘x’ ‘drop2nd’

Undocumented arguments in documentation object 'gafs.default'
  ‘...’
Documented arguments not in \usage in documentation object 'gafs.default':
  ‘list()’

Documented arguments not in \usage in documentation object 'gafs_initial':
  ‘population’ ‘fitness’ ‘parent’ ‘parents’ ‘r’ ‘q’ ‘k’

Documented arguments not in \usage in documentation object 'icr.formula':
  ‘x’ ‘y’ ‘object’ ‘newdata’

Documented arguments not in \usage in documentation object 'knn3':
  ‘formula’ ‘data’ ‘subset’ ‘na.action’ ‘k’ ‘y’ ‘train’ ‘test’ ‘cl’ ‘l’
  ‘prob’ ‘use.all’

Documented arguments not in \usage in documentation object 'knnreg':
  ‘formula’ ‘data’ ‘subset’ ‘na.action’ ‘k’ ‘y’ ‘train’ ‘test’
  ‘use.all’

Documented arguments not in \usage in documentation object 'lift':
  ‘data’ ‘class’ ‘subset’ ‘lattice.options’ ‘cuts’ ‘labels’ ‘plot’
  ‘values’

Documented arguments not in \usage in documentation object 'maxDissim':
  ‘u’

Documented arguments not in \usage in documentation object 'modelLookup':
  ‘regex’ ‘pkg’

Documented arguments not in \usage in documentation object 'nearZeroVar':
  ‘y’ ‘index’

Documented arguments not in \usage in documentation object 'nullModel':
  ‘y’ ‘object’ ‘newdata’ ‘type’

Documented arguments not in \usage in documentation object 'oneSE':
  ‘tol’

Documented arguments not in \usage in documentation object 'pcaNNet.default':
  ‘formula’ ‘weights’ ‘data’ ‘subset’ ‘na.action’ ‘contrasts’ ‘object’
  ‘newdata’ ‘type’

Documented arguments not in \usage in documentation object 'plot.rfe':
  ‘data’ ‘output’ ‘mapping’ ‘environment’

Documented arguments not in \usage in documentation object 'plot.train':
  ‘data’ ‘output’ ‘highlight’ ‘mapping’ ‘environment’

Documented arguments not in \usage in documentation object 'plot.varImp.train':
  ‘data’ ‘mapping’ ‘environment’

Documented arguments not in \usage in documentation object 'plsda':
  ‘y’ ‘ncomp’ ‘probMethod’ ‘prior’ ‘object’ ‘newdata’ ‘type’

Documented arguments not in \usage in documentation object 'postResample':
  ‘data’ ‘lev’ ‘model’ ‘formula’ ‘na.rm’ ‘x’

Documented arguments not in \usage in documentation object 'prcomp.resamples':
  ‘what’ ‘dims’

Documented arguments not in \usage in documentation object 'preProcess':
  ‘method’ ‘thresh’ ‘pcaComp’ ‘na.remove’ ‘object’ ‘newdata’ ‘k’
  ‘knnSummary’ ‘outcome’ ‘fudge’ ‘numUnique’ ‘verbose’

Undocumented arguments in documentation object 'predict.train'
  ‘...’
Documented arguments not in \usage in documentation object 'predict.train':
  ‘list()’ ‘models’ ‘testX’ ‘testY’ ‘unkX’ ‘unkOnly’ ‘verbose’

Undocumented arguments in documentation object 'print.train'
  ‘...’
Documented arguments not in \usage in documentation object 'print.train':
  ‘list()’

Documented arguments not in \usage in documentation object 'recall':
  ‘reference’ ‘relevant’ ‘beta’ ‘na.rm’

Documented arguments not in \usage in documentation object 'resamples':
  ‘modelNames’ ‘object’ ‘metric’ ‘decreasing’ ‘FUN’ ‘row.names’
  ‘optional’

Documented arguments not in \usage in documentation object 'rfe':
  ‘y’ ‘testX’ ‘testY’ ‘sizes’ ‘metric’ ‘maximize’ ‘rfeControl’ ‘object’
  ‘size’ ‘newdata’ ‘label’ ‘seeds’

Undocumented arguments in documentation object 'safs.default'
  ‘...’
Documented arguments not in \usage in documentation object 'safs.default':
  ‘list()’

Documented arguments not in \usage in documentation object 'safsControl':
  ‘genParallel’

Documented arguments not in \usage in documentation object 'safs_initial':
  ‘x’ ‘old’ ‘new’ ‘iteration’ ‘number’

Undocumented arguments in documentation object 'sbf'
  ‘...’
Documented arguments not in \usage in documentation object 'sbf':
  ‘list()’ ‘y’ ‘form’ ‘data’ ‘subset’ ‘na.action’ ‘contrasts’
  ‘sbfControl’ ‘object’ ‘newdata’

Documented arguments not in \usage in documentation object 'sensitivity':
  ‘reference’ ‘positive’ ‘negative’ ‘prevalence’ ‘na.rm’

Undocumented arguments in documentation object 'train'
  ‘...’
Documented arguments not in \usage in documentation object 'train':
  ‘list()’ ‘y’ ‘form’ ‘data’ ‘weights’ ‘subset’ ‘na.action’ ‘contrasts’
  ‘method’ ‘preProcess’ ‘metric’ ‘maximize’ ‘trControl’ ‘tuneGrid’
  ‘tuneLength’

Documented arguments not in \usage in documentation object 'twoClassSim':
  ‘factors’ ‘class’

Documented arguments not in \usage in documentation object 'varImp':
  ‘useModel’ ‘nonpara’ ‘scale’ ‘numTrees’ ‘threshold’ ‘data’ ‘value’
  ‘surrogates’ ‘competes’ ‘estimate’ ‘drop’ ‘cuts’ ‘weights’ ‘lambda’

Documented arguments not in \usage in documentation object 'xyplot.resamples':
  ‘variables’ ‘panelRange’ ‘conf.level’

Functions with \usage entries need to have the appropriate \alias
entries, and all their arguments documented.
The \usage entries must correspond to syntactically valid R code.
See chapter ‘Writing R documentation files’ in the ‘Writing R
Extensions’ manual.
S3 methods shown with full name in documentation object 'plot.gafs':
  ‘plot.gafs’

S3 methods shown with full name in documentation object 'plot.rfe':
  ‘plot.rfe’

S3 methods shown with full name in documentation object 'plot.train':
  ‘plot.train’

S3 methods shown with full name in documentation object 'summary.bagEarth':
  ‘summary.bagEarth’

The \usage entries for S3 methods should use the \method markup and not
their full name.
See chapter ‘Writing R documentation files’ in the ‘Writing R
Extensions’ manual.
```
Can you take a look at these and update the `roxygen` comments?

**Comment by JackStat** [comment]
Oh wow. I will get on it tomorrow... Sorry about that.

**Pull request merged**